engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.h

```diff
@@ -369,19 +353,23 @@ class PartitionKernel : public kernel {
 							continue;
 						}
 
-						metadata.add_value(ral::cache::WORKER_IDS_METADATA_LABEL, dest_node.id());
-						metadata.add_value(ral::cache::CACHE_ID_METADATA_LABEL, "output_" + std::to_string(part_ids[i]) );
-						
-						node_count[dest_node.id()]++;
-						output_cache->addCacheData(std::unique_ptr<ral::cache::GPUCacheData>(new ral::cache::GPUCacheDataMetaData(table_view.clone(), metadata)),"",true);
+						send_message(std::move(table_view.clone()),
+							"true", //specific_cache
+							"output_" + std::to_string(part_ids[i]), //cache_id
+							dest_node.id(), //target_id
+							"", //total_rows
+							"", //message_id_prefix
+							true); //always_add
+
+						increment_node_count(dest_node.id(), 0);
 					}
 
 					for (auto i = 0; i < partitions.size(); i++) {
 						auto & partition = partitions[i];
```

Comment:
we should make `send_message` know what to do with sending to `self_node` that way we dont need this sort of logic for messages not actually being sent anywhere

Reply:
This change is not as simple as I thought, I mean, this pattern could apparently be replaced by a scatter but a bit more general.